### ISSUE TITLE
LSH Implementation with TFIDF Dense Matrix

### ISSUE THREAD

**Issue by girishmt4** [FALCONN-LIB/FALCONN#101]
I am currently working on Documents similarity project. We are processing text documents to generate TFIDF Vectors for each document in the corpus. In a nutshell, we are working with DENSE DATA with the documents being the data points and TFIDF values of the terms occuring in the document as their features.
We succeeded in implementing LSH with sparse data but it's not quite efficient. 
Is it possible to use FALCONN with dense data for LSH implementation?

**Comment by ludwigschmidt** [comment]
Yes, FALCONN supports dense data. In fact, the support for dense data is better than for sparse data. But if your data is very high-dimensional, the dense approach might not be efficient. What dimension do you work with?

**Comment by girishmt4** [comment]
I am currently working with a dataset that stores the TF-IDF values for only those terms that occur in the particular document. So, every point will have different dimension.
What is your say on this?

**Comment by ludwigschmidt** [comment]
In that case, using a sparse representation might be better.

**Comment by girishmt4** [comment]
can you explain the reason behind that? I am still wondering why sparse representation can perform better than the dense one!

**Comment by ludwigschmidt** [comment]
With a dense representation, the code will be performing many unnecessary multiplications with zero.

**Comment by girishmt4** [comment]
I am extremely sorry! My understanding of dense and sparse representations was reversed. 
I am currently using 20NewsGroup's training dataset of 10000 documents. 
I am using Sparse Representation now, which is a typedef of a vector of std::pair where the pair contains the index and the non-zero value itself if I am not wrong.
I am having issues while finding the center of mass since the size of every SparseVector will be different.
